### ISSUE TITLE
Server render mismatch

### ISSUE THREAD

**Issue by ianb** [mozilla-services/pageshot#1107]
Error:

```
Warning: React attempted to reuse markup in a container but the checksum was invalid. This generally means that you are using server rendering and the markup generated on the server was not what the client was expecting. React injected new markup to compensate which works but you have lost many of the benefits of server rendering. Instead, figure out why the markup being generated is different on the client or server:
 (client) nk Wilczek, review: &#x27;worth the effort&#x27;</span><span data-reactid=".1j1g3kj8hz4.1.1.2"> </span></span><div class="shot-subtitle" data-reactid=".1j1g3kj8hz4.1.2"><span data-reactid=".1j1g3kj8hz4.1.2.0">Saved from </span><a class="subheading-link" href="/redirect?to=http%3A%2F%2Fwww.telegraph.co.uk%2Fculture%2Fbooks%2Fbookreviews%2F11773491%2FA-Beautiful-Question-by-Frank-Wilczek.html" data-reactid=".1j1g3kj8hz4.1.2.1">telegraph.co.uk</a><img height="16" width="16" style="margin-right:7px;margin-left:7px;position:relative;top:4px;" src="/static/img/clock.svg?rev=" data-reactid=".1j1g3kj8hz4.1.2.2"><span title="Feb 2 2016, 4pm" data-reactid=".1j1g3kj8hz4.1.2.3">6 minutes ago</span><span data-reactid=".1j1g3kj8hz4.1.2.4"> </span><span data-reactid=".1j1g3kj8hz4.1.2.5"><span data-reactid=".1j1g3kj8hz4.1.2.5.0">– </span><button style="border:1px solid #d4d4d4;border-radius:3px;background-color:transparent;padding:3px 10px;font-size:11px;color:#858585;" data-reactid=".1j1g3kj8hz4.1.2.5.1"><span data-reactid=".1j1g3kj8hz4.1.2.5.1.0"><span data-reactid=".1j1g3kj8hz4.1.2.5.1.0.0">expires</span><span data-reactid=".1j1g3kj8hz4.1.2.5.1.0.1"> </span><span title="Mar 2 2016, 4pm" data-reactid=".1j1g3kj8hz4.1.2.5.1.0.2">in 14 days</span></span></button></span></div><div class="more-shot-actions" data-reactid=".1j1g3kj8hz4.1.3"><button class="share-button" data-reactid=".1j1g3kj8hz4.1.3.1">Share</button><button class="trash-button" data-reactid=".1j1g3kj8hz4.1.3.2"><img src="/static/img/garbage-bin.png?rev=" data-reactid=".1j1g3kj8hz4.1.3.2.0"></button></div></div><iframe width="100%" height="7395" id="frame" src="//pageshotcontent.dev.mozaws.net/content/afsiWwxXpNt3Jp4Z/www.telegraph.co.uk" style="background-color:#fff;" data-reactid=".1j1g3kj8hz4.5"></iframe><div class="pageshot-footer" data-reactid=".1j1g3kj8hz4.6"><a href="https://github.com/mozilla-services/pageshot" data-reactid=".1j1g3kj8hz4.6.0">PageShot</a><span data-reactid=".1j1g3kj8hz4.6.1"> — </span><a href="https://github.com/moz
 (server) nk Wilczek, review: 'worth the effort'</span><span data-reactid=".1j1g3kj8hz4.1.1.2"> </span></span><div class="shot-subtitle" data-reactid=".1j1g3kj8hz4.1.2"><span data-reactid=".1j1g3kj8hz4.1.2.0">Saved from </span><a class="subheading-link" href="/redirect?to=http%3A%2F%2Fwww.telegraph.co.uk%2Fculture%2Fbooks%2Fbookreviews%2F11773491%2FA-Beautiful-Question-by-Frank-Wilczek.html" data-reactid=".1j1g3kj8hz4.1.2.1">telegraph.co.uk</a><img style="margin-right:7px;margin-left:7px;position:relative;top:4px;" src="/static/img/clock.svg?rev=" data-reactid=".1j1g3kj8hz4.1.2.2" height="16" width="16"><span title="Feb 2 2016, 9pm" data-reactid=".1j1g3kj8hz4.1.2.3">6 minutes ago</span><span data-reactid=".1j1g3kj8hz4.1.2.4"> </span><span data-reactid=".1j1g3kj8hz4.1.2.5"><span data-reactid=".1j1g3kj8hz4.1.2.5.0">–&nbsp;</span><button style="border:1px solid #d4d4d4;border-radius:3px;background-color:transparent;padding:3px 10px;font-size:11px;color:#858585;" data-reactid=".1j1g3kj8hz4.1.2.5.1"><span data-reactid=".1j1g3kj8hz4.1.2.5.1.0"><span data-reactid=".1j1g3kj8hz4.1.2.5.1.0.0">expires</span><span data-reactid=".1j1g3kj8hz4.1.2.5.1.0.1"> </span><span title="Mar 2 2016, 9pm" data-reactid=".1j1g3kj8hz4.1.2.5.1.0.2">in 14 days</span></span></button></span></div><div class="more-shot-actions" data-reactid=".1j1g3kj8hz4.1.3"><button class="share-button" data-reactid=".1j1g3kj8hz4.1.3.1">Share</button><button class="trash-button" data-reactid=".1j1g3kj8hz4.1.3.2"><img src="/static/img/garbage-bin.png?rev=" data-reactid=".1j1g3kj8hz4.1.3.2.0"></button></div></div><iframe id="frame" src="//pageshotcontent.dev.mozaws.net/content/afsiWwxXpNt3Jp4Z/www.telegraph.co.uk" style="background-color:#fff;" data-reactid=".1j1g3kj8hz4.5" height="7395" width="100%"></iframe><div class="pageshot-footer" data-reactid=".1j1g3kj8hz4.6"><a href="https://github.com/mozilla-services/pageshot" data-reactid=".1j1g3kj8hz4.6.0">PageShot</a><span data-reactid=".1j1g3kj8hz4.6.1"> — </span><a href="https://github.com/mozilla-
```

In page http://pageshot.dev.mozaws.net/afsiWwxXpNt3Jp4Z/www.telegraph.co.uk

**Comment by fzzzy** [comment]
I think we can fix this by replacing ' with &#x27; on the server. This happens sometimes when the browser silently changes stuff between setting something and accessing it.

**Comment by ianb** [comment]
Huh, I can't reproduce this locally, not sure what's going on here.

**Comment by ianb** [comment]
I'm still seeing these errors frequently on the site.  git-rev issues maybe?  Not sure, needs investigation.

**Comment by fzzzy** [comment]
Luckily this just wastes cpu and is otherwise harmless.

**Comment by ianb** [comment]
I'm pretty sure this is fixed with #1500 / 9428fda10894f804b40fee12751da84b8cbe9c76
